lib/data/adapters/activity_isar_adapter.dart

```diff
@@ -18,11 +18,13 @@ class ActivityIsarAdapter extends BaseAdapter<Activity, ActivityModel> {
     return Activity(
       name: m.name,
       sport: m.sport,
-      // TODO : convert startTime and stopTime back to their original timezone
       startTime: m.startTime,
+      // TODO : handle timezone using startTimeZoneOffsetMinutes and Name
       stopTime: m.stopTime,
       distanceInMeters: m.distanceInMeters,
-      duration: Duration(microseconds: m.durationInMicroSeconds),
+      duration: m.durationInMicroSeconds != null
+          ? Duration(microseconds: m.durationInMicroSeconds!)
+          : null,
```

Comment:
this is a personal preference but I think avoiding `!` operators in ternary conditions is more readable

```suggestion
      duration: m.durationInMicroSeconds == null
          ? null
          : Duration(microseconds: m.durationInMicroSeconds!),
```
again, just a personal preference if you prefer having the non-null result as the first part of the ternary condition then fine